src/main/webapp/components/results/result-index-cards.js

```diff
@@ -8,12 +8,34 @@ import useAnchorEl from '../../react-hooks/use-anchor-el'
 import { ResultListInteraction } from '../lists/result-list-interaction'
 import { ExportAction } from '../result-export/result-export-action'
 import Divider from '@material-ui/core/Divider'
+import Checkbox from '@material-ui/core/Checkbox'
 
 export default props => {
-  const { results, setLists, lists } = props
+  const {
+    results,
+    setLists,
+    lists,
+    resultsToExport,
+    setResultsToExport,
+  } = props
   const [selectedResult, setSelectedResult] = React.useState(null)
   const [anchorEl, handleOpen, handleClose, isOpen] = useAnchorEl()
 
+  const modifySelectedResults = result => {
+    const index = resultsToExport.indexOf(
+      item => result.attributes.id === item.attributes.id
+    )
+    if (index === -1) {
+      const updatedResults = resultsToExport
```

Comment:
I tried doing 
`setResultsToExport([...resultsToExport, result])` here but the sate was never getting updated. That's what I did it this way. Not sure if I was doing something wrong.

Reply:
I think this is because of the React.memo call on results. The list of results won't re-render unless results change: https://github.com/connexta/revelio/blob/master/src/main/webapp/components/results/result-index-cards.js#L59

When you update the state to a new array, results are the same, the list of results doesn't re-render, and so the individual results still have access to the old array. That's why pushing to that array works (but we shouldn't be doing it that way). One solution would be to add resultsToExport to the react memo dependency array so that it re-renders on changes to that array.

That being said, if you go the route of using the selection interface internally in the exportaction and checkbox, you won't need to worry about it because they will re-render independently of the result list (I think this is the best option).

Reply:
After investigating this for a while, I've decided to break that out into another pr. Since the export calls to ddf require the list of sources to each result, I would need to add the sourceId the selectioninterface as opposed to just the ids. This would require touching all the components that currently use the selectionInterface.